AnaProd/anaTupleProducer.py

```diff
@@ -189,24 +196,31 @@ def createAnatuple(inFile, outFile, config, sample_name, anaCache, snapshotOptio
             weight_branches.extend(dfw.Apply(Corrections.trg.getTrgSF, trigger_class.trigger_dict.keys(), is_central and compute_unc_variations))
             weight_branches.extend(dfw.Apply(Corrections.btag.getSF,is_central and compute_unc_variations))
             dfw.colToSave.extend(weight_branches)
-        #print("going to evaluate the report")
-        report = dfw.df.Report()
-        if print_cutflow:
-            report.Print()
+        reports.append(dfw.df.Report())
         varToSave = Utilities.ListToVector(dfw.colToSave)
-        #print(f"saving the tree Events{suffix}")
-        dfw.df.Snapshot(f"Events{suffix}", outFile, varToSave, snapshotOptions)
-        snapshotOptions.fMode = "UPDATE"
-        histReport = ReportTools.SaveReport(report.GetValue(), reoprtName=f"Report{suffix}")
-        outputRootFile= ROOT.TFile(outFile, "UPDATE", "", compression_settings)
-        outputRootFile.WriteTObject(histReport, f"Report{suffix}", "Overwrite")
+        #print(f"snapshot flazy = {snapshotOptions.fLazy}")
+        outFileName = f"{outDir}Events{suffix}.root"
+        outfilesNames.append(outFileName)
+        if os.path.exists(outFileName):
+            os.remove(outFileName)
+        snaps.append(dfw.df.Snapshot(f"Events", outFileName, varToSave, snapshotOptions))
+        hist_time = ROOT.TH1D(f"time{suffix}", f"time{suffix}", 1, 0, 1)
+        end_time = datetime.datetime.now()
```

Comment:
hist time should be one for all (e.g. you can save it in the first file) and estimated after RunGraphs